crates/sui-graphql-rpc/src/context_data/pg_backend.rs

```diff
@@ -457,19 +459,60 @@ impl GenericQueryBuilder<Pg> for PgQueryBuilder {
         }
 
         // Filters on the package and/ or module that emitted some event
-        if let Some(p) = filter.emitting_package {
-            query = query.filter(events::dsl::package.eq(p.into_vec()));
+        if let Some(pm) = filter.emitting_module {
+            let format = "package[::module]";
+            let parts: Vec<_> = pm.splitn(2, "::").collect();
+
+            if parts.iter().any(|&part| part.is_empty()) {
+                return Err(DbValidationError::InvalidType(
+                    TypeFilterError::MissingComponents(pm, format).to_string(),
```

Comment:
nit: this is going to complain that the thing that has a bad format is a type when in fact it is a module.